app/src/main/java/edu/uc/kovaciad/slicetracker/ui/main/MainViewModel.kt

```diff
@@ -10,10 +10,14 @@ import com.google.firebase.firestore.FirebaseFirestoreSettings
 import edu.uc.kovaciad.slicetracker.dto.*
 
 class MainViewModel : ViewModel() {
-    var firestore: FirebaseFirestore = FirebaseFirestore.getInstance()
+    private var firestore: FirebaseFirestore = FirebaseFirestore.getInstance()
 
     init {
         firestore.firestoreSettings = FirebaseFirestoreSettings.Builder().build()
+        createListeners()
```

Comment:
Nesting all listener creation functions into one function for better convenience, reusability, and readability